js/src/util/config.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * --------------------------------------------------------------------------
+ * Bootstrap (v5.0.0-beta3): util/backdrop.js
+ * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
+ * --------------------------------------------------------------------------
+ */
+
+import { typeCheckConfig } from './index'
+
+class Config {
+  constructor(config) {
+    this._config = this._getConfig(config)
+  }
+
+  static get NAME() {
+    throw new Error('You have to implement the static method "NAME", for each component!')
+  }
+
+  static get Default() {
+    return {}
+  }
+
+  static get DefaultType() {
+    return {}
+  }
+
+  _getConfig(config) {
+    config = this._mergeConfigObj(config)
+    this._typeCheckConfig(config)
```

Comment:
Why not use `typeCheckConfig()` directly here and remove the wrapper? BTW, I think it might make sense to move `typeCheckConfig` from index.js to config.js, maybe later.

Reply:
I am on experiment period yet. ;)

>BTW, I think it might make sense to move typeCheckConfig from index.js to config.js, maybe later.

this exactly I have in mind